engine/include/windowing/window.hpp

```diff
@@ -4,61 +4,81 @@
 
 #include "core.hpp"
 
+#include <memory>
 #include <string>
 
 // TODO: Replace this with a proper implementation
 
 // Start Forward declarations
 struct GLFWwindow;
 
-namespace vk
+// End Forward declarations
+
+namespace gen
 {
-    class UniqueInstance;
-    class UniqueSurfaceKHR;
-}
 
-// End Forward declarations
+	class Window
+	{
+	public:
+		// Values mirrored from GLFW
+		// https://www.glfw.org/docs/3.3/glfw3_8h.html#a2315b99a329ce53e6a13a9d46fd5ca88
+		enum class CursorMode
+		{
+			Normal = 0x00034001,
+			Hidden = 0x00034002,
+			Disabled = 0x00034003
+		};
 
-namespace gen {
+		Window(int width, int height, std::string title);
+		~Window();
 
-class Window
-{
-public:
-    Window(int w, int h, std::string title);
-    ~Window();
+		Window(const Window &)			   = delete;
+		Window & operator=(const Window &) = delete;
+		Window(Window &&)				   = default;
+		Window & operator=(Window &&)	   = default;
+
+		// Member functions
+
+		GEN_NODISCARD bool shouldClose() const;
+		static void pollEvents();
 
-	Window(const Window&) = delete;
-	Window& operator=(const Window&) = delete;
+		// Getters
 
-	// Member functions
+		GEN_NODISCARD int getWidth() const { return m_width; }
+		GEN_NODISCARD int getHeight() const { return m_height; }
+		GEN_NODISCARD const std::string & getTitle() const { return m_title; }
```

Comment:
Should return `string_view`.

Reply:
Why should we return a string_view instead of a string? Wouldn't that create a return of reference to a temporary object?